x-pack/plugins/ml/server/routes/trained_models.ts

```diff
@@ -652,4 +654,71 @@ export function trainedModelsRoutes({
         }
       })
     );
+
+  /**
+   * @apiGroup TrainedModels
+   *
+   * @api {post} /internal/ml/trained_models/model_downloads Gets available models for download
+   * @apiName GetTrainedModelDownloadList
+   * @apiDescription Gets available models for download.
+   */
+  router.versioned
+    .get({
+      path: `${ML_INTERNAL_BASE_PATH}/trained_models/model_downloads`,
+      access: 'internal',
+      options: {
+        tags: ['access:ml:canGetTrainedModels'],
+      },
+    })
+    .addVersion(
+      {
+        version: '1',
+        validate: {
+          request: {
+            query: modelDownloadsQuery,
+          },
+        },
+      },
+      routeGuard.fullLicenseAPIGuard(async ({ mlClient, request, response, client }) => {
+        try {
+          const nodesInfoResponse =
+            await client.asInternalUser.transport.request<estypes.NodesInfoResponseBase>({
+              method: 'GET',
+              path: `/_nodes/ml:true/os`,
+            });
+
+          let osName: string | undefined;
+          let arch: string | undefined;
```

Comment:
A key case to consider is the case of zero ML nodes but running in Cloud, where it's likely that autoscaling will be enabled. In this case we want to recommend the Linux x86_64 optimized model.

Reply:
Added a check for running in the cloud [here](https://github.com/elastic/kibana/pull/167000/commits/fe61b68b36cf17f1902a90335da62e30ced81269).

@jgowdyelastic could you please check if it's the correct approach?

Reply:
We can check for lazy ML nodes. This is what we do in other places where need to know that nodes will become available.
`getLazyMlNodeCount` is a server side function you can use.

>could you please check if it's the correct approach?

yes, checking for `cloud.cloudId` is what we use.

Reply:
I'm not sure if we need to check for lazy ML nodes. Only the architecture is important for determining the model type. I guess the current check should be sufficient.

Reply:
Yes, people should still be allowed to _download_ the model, even if they've used up all their lazy nodes. For trials people only get one. If it's Cloud then the best model to download is the optimized linux x86_64 one, so we should always let people download that if they want to. After downloading it they might find out they need to stop other jobs or models in order to start it.